pkgs/by-name/do/doomsday-engine/3.x.nix

```
@@ -0,0 +1,66 @@
+{
+  callPackage,
+  fetchFromGitea,
+  autoPatchelfHook,
+  zlib,
+
+  symlinkJoin,
+  fetchFromGitHub,
+  glbinding,
+  assimp,
+  xorg,
+}:
+callPackage (import ./generic.nix {
+  pname = "doomsday-engine3";
+  version = "3.0-unstable-2021-08-07";
+
+  src = fetchFromGitea {
+    domain = "git.skyjake.fi";
+    owner = "doomsday";
+    repo = "engine";
+    rev = "db4bb26129d6fca8dcd58ba2a058dc73eaa3ef86";
+    hash = "sha256-tox5c4UPfq5tNBLkD/ebHMiWpDN18WDuZRKzFzekwJQ=";
+  };
+
+  gamekitPath = "libs/gamekit/libs";
+
+  nativeBuildInputs = [ autoPatchelfHook ];
+  buildInputs = [ zlib ];
+
+  preConfigure =
+    let
+      # Doomsday Engine 3.x currently depends on some *really* old deps...
+      # They have to be symlinked to the right path, too...
+      vendoredDeps = symlinkJoin {
+        name = "doomsday-vendored-deps";
+        paths = [
+          (glbinding.overrideAttrs (prev: {
+            version = "2.1.4";
+            src = fetchFromGitHub {
+              inherit (prev.src) owner repo;
+              rev = "v2.1.4";
+              hash = "sha256-n/7xJpYgjW3iwjlh7aMrHSvF9umUYSszP6x/HMUdLPo=";
+            };
```

could also use `rec` here but we all know that debate by now
```suggestion
          (glbinding.overrideAttrs (final: prev: {
            version = "2.1.4";
            src = fetchFromGitHub {
              inherit (prev.src) owner repo;
              rev = "v${final.version}";
              hash = "sha256-n/7xJpYgjW3iwjlh7aMrHSvF9umUYSszP6x/HMUdLPo=";
            };
```